lib/read-from-url.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+const tmp = require('tmp');
+const fetch = require('minipass-fetch');
+const fs = require('fs');
+const debug = require('debug')('supported:read-from-url');
+
+const RAW_CONTENT_HOST = 'raw.githubusercontent.com';
+
+async function setupProjectPath(_url, rawFileHost) {
+  let tempDir = tmp.dirSync();
+
+  let url = new URL(_url);
+  url.host = rawFileHost || RAW_CONTENT_HOST;
+  if (url.pathname.indexOf('tree') == -1) {
+    url.pathname += '/master/package.json';
+  }
+
+  debug(`packageJSON url: ${url.toString()}`);
+  const packageJSON = await fetch(url.toString()).then(async request => {
```

Comment:
can we use `async` & `await` and avoid the `then` entirely? It should help with debugging/improved stacktraces.